vault/quotas/quotas.go

```diff
@@ -543,6 +552,20 @@ func (m *Manager) SetEnableRateLimitResponseHeaders(val bool) {
 	m.config.EnableRateLimitResponseHeaders = val
 }
 
+// SetRateLimitExemptPaths updates the rate limit exempt paths in the Manager's
+// configuration in addition to updating the path manager. Every call to
+// SetRateLimitExemptPaths will wipe out the existing path manager and set the
+// paths based on the provided argument.
+func (m *Manager) SetRateLimitExemptPaths(vals []string) {
+	if vals == nil {
+		vals = []string{}
+	}
+
+	m.config.RateLimitExemptPaths = vals
+	m.rateLimitPathManager = pathmanager.New()
```

Comment:
I'm not sure if this is the race condition you referred to already, but it seems like this change can occur concurrently with requests coming in. Does pathmanager have its own internal mutex?

Might be slightly better to create the manager locally, add the paths to it, and then swap it into rateLimitPathManager.

Reply:
I'm not able to find how any of the manager's attributes are protected against concurrent access? It seems like SetEnableRateLimitResponseHeaders and SetEnableRateLimitAuditLogging have the same problem, or something is protecting all three that I'm not seeing.

Reply:
> I'm not sure if this is the race condition you referred to already, but it seems like this change can occur concurrently with requests coming in. Does pathmanager have its own internal mutex?
> 
> Might be slightly better to create the manager locally, add the paths to it, and then swap it into rateLimitPathManager.

No, it's not the race condition I was referring to based on what I see in the unit tests. However, the point you raise is not specific to the path manager. It's applicable to all three configurations (e.g. enabling/disabling response headers can happen mid-request). This is why I was considering adding a lock around the entire manager's use in `Core`, not just for the path manager.

Reply:
> I'm not able to find how any of the manager's attributes are protected against concurrent access?

They're not! I think we need to dig further into this. cc @vishalnayak

Reply:
Hey there, pathmanager does have an internal lock of its own. Quota manager also have a lock of its own to protect its attributes. Some functions don't use it directly because the functions from where they are called from are holding the lock needed to protect them.

We might want to look at all the callers and see if they are holding locks properly (in this PR), and probably update the comment on the functions that the callers should hold the lock.

Reply:
Taking a quick look I don't see any protection against rate limit application during a config or quota update.

i.e. `ApplyRateLimitQuota` can be called at the same time as `handleRateLimitQuotasUpdate`/`handleQuotasConfigUpdate`. Since the `Manager` has a RW mutex, why not just use that for _all_ methods @vishalnayak?

Reply:
Going to resolve this. Spoke with @vishalnayak and it seems this isn't an issue.